addons/binding/org.openhab.binding.roomba/src/main/java/org/openhab/binding/roomba/discovery/RoombaDiscoveryService.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    /**
+     * Scans for roombas by spawning a search thread
+     */
+    @Override
+    protected void startScan() {
+        logger.info("Starting discovery for iRobot Roombas");
+        scheduler.execute(searchRunnable);
+    }
+
+    // Runnable for search adapter
+
+    private Runnable searchRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            logger.debug("Start adapter discovery ");
+            logger.trace("Send broadcast message");
+            DatagramSocket localSocket = null;
+
+            try {
+                localSocket = new DatagramSocket();
+                localSocket.setBroadcast(true);
+                localSocket.setSoTimeout(10000); // Listen 10 seconds
+
+                byte[] sendData = BROADCAST_MESSAGE.getBytes();
+
+                DatagramPacket sendPacket = new DatagramPacket(sendData, sendData.length,
+                        InetAddress.getByName(BROADCAST_ADDRESS), BROADCAST_PORT);
+                localSocket.send(sendPacket);
+
+                byte[] receiveBuffer = new byte[1000];
+
+                // Listen for answer
+
+                DatagramPacket receivePacket = new DatagramPacket(receiveBuffer, receiveBuffer.length);
+                localSocket.receive(receivePacket);
+                String receiveMessage = new String(receivePacket.getData()).trim();
+                String receiveIP = receivePacket.getAddress().getHostAddress();
+                int receivePort = receivePacket.getPort();
+                logger.trace("Received Message: {}  ", receiveMessage);
+                logger.trace("Received from Host: {}", receiveIP);
+                logger.trace("Received from Port: {}", receivePort);
+
+                // Parse the response (json) into its POJO equivalent
+                Gson gson = new Gson();
+                RoombaDiscoveryResult discoveryResult = gson.fromJson(receiveMessage, RoombaDiscoveryResult.class);
+
+                // Add the roomba as a discovery result
+                addRoomba(discoveryResult);
+            } catch (IOException e) {
+                logger.info("Unable to find any Roomba!", e);
+            } finally {
+                try {
+                    if (localSocket != null) {
+                        localSocket.close();
+                    }
+                } catch (Exception e) {
+                    logger.error("Failed to close socket!", e);
+                }
+            }
+        }
+
+    };
+
+    /**
+     * Adds a single Roomba from its parsed response.
+     *
+     * @param roombaDiscoveryResult The discovery result for the Roomba
+     */
+    private void addRoomba(RoombaDiscoveryResult roombaDiscoveryResult) {
+        if (roombaDiscoveryResult == null) {
+            logger.warn("Received an unexpected null response from the Roomba. Ignoring.");
+            return;
+        }
+
+        Map<String, Object> properties = new HashMap<>(3);
```

Comment:
why initialize it with a length 3 and add a single element, maybe just use the default constructor.